tests/unittest/test_helpers.py

```diff
@@ -166,6 +173,981 @@ def test_merge_configurations(self):
             == expected_merged_options.strip()
         )
 
+    def test_parse_conf_overrides(self):
+        test_id = str(uuid.uuid4())
+
+        username = str(uuid.uuid4())
+        namespace = str(uuid.uuid4())
+        kubeconfig = str(uuid.uuid4())
+        context = str(uuid.uuid4())
+
+        os.environ["CONF_OVERRIDE_CONTEXT"] = context
+
+        conf_list = [
+            f"spark.kubernetes.namespace={namespace}",
+            f"spark.kubernetes.authenticate.driver.serviceAccountName={username}",
+            f"{helpers.constants.OPTION_SPARK_DRIVER_EXTRA_JAVA_OPTIONS}=-Dscala.shell.histfile={test_id} -Dkubeconfig={kubeconfig} -Dcontext=$CONF_OVERRIDE_CONTEXT",
+        ]
+
+        overrides = helpers.utils.parse_conf_overrides(conf_list)
+
+        assert overrides.get("spark.kubernetes.namespace") == namespace
+        assert (
+            overrides.get("spark.kubernetes.authenticate.driver.serviceAccountName")
+            == username
+        )
+        assert (
+            overrides.get(helpers.constants.OPTION_SPARK_DRIVER_EXTRA_JAVA_OPTIONS)
+            == f"-Dscala.shell.histfile={test_id} -Dkubeconfig={kubeconfig} -Dcontext={context}"
+        )
+
+    @patch("sys.exit")
+    def test_parse_conf_overrides_invalid(self, mock_sys_exit):
+        mock_sys_exit.return_value = 0
+
+        conf_list = [
+            "spark.kubernetes.namespace",
+            "spark.kubernetes.authenticate.driver.serviceAccountName",
+        ]
+
+        overrides = helpers.utils.parse_conf_overrides(conf_list)
+
+        mock_sys_exit.assert_any_call(helpers.constants.EXIT_CODE_BAD_CONF_ARG)
```

Comment:
Very very good to tests happy path but most importantly when things go wrong!!!!!

:+1: